File: problems/z/zanzibar/zanzibar.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int t, turtles, turtles_abroad, previous_turtles;

    cin >> t;

    for (int i=0; i<t; i++) {
        cin >> turtles;
        turtles_abroad = 0;
        previous_turtles = 1;
        while (turtles != 0) {
            if (turtles > (previous_turtles * 2))
                turtles_abroad += turtles - (previous_turtles * 2);
            previous_turtles = turtles;

            cin >> turtles;
        }
        cout << turtles_abroad << endl;
    }
}

File: problems/z/zanzibar/zanzibar.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    t = int(input())

    for _ in range(t):
        list_turtles = [int(val) for val in input().split()]
        turtles_abroad = 0
        previous_turtles = 1
        for turtles in list_turtles[:-1]:
            if turtles > (previous_turtles * 2):
                turtles_abroad += turtles - (previous_turtles * 2)
            previous_turtles = turtles
        print(turtles_abroad)

if __name__ == '__main__':
    main()